system/jlib/jmetrics.cpp

```diff
@@ -153,6 +153,13 @@ bool MetricsManager::addMetric(const std::shared_ptr<IMetric> &pMetric)
     bool rc = false;
     std::string name = pMetric->queryName();
 
+    // Remove unwanted characters from input name
```

Comment:
A couple of comments on this code:
i) It would possibly be better to not perform if the string doesn't contain any of those characters
ii) I think you could remove all characters in one go and avoid multiple passes/copies of the name.  Something like
name.erase(std::remove_if(name.begin(), name.end(), [](unsigned char x) { strchr(removeChars, x) != nullptr }));
iii) Shouldn't this be the responsibility of the caller to pass a clean name?

Reply:
@ghalliday 

The case that caused the bug to be written is most likely because the ESP name from a helm chart (values.yaml probably) has a dash in it. The generated ESP service code for execution profiling uses the ESP process name when constructing metric names. 

I considered adding code to the generated code to clean up the name prior to registering the metric, however a comment by Tony in the Jira suggested cleaning it up in the metrics code. The advantage of cleaning in the metrics code is it relieves all components from worrying about using names sourced from a configuration file (where a dash or other character may be legal) from having to clean the name prior to registering a metric.

If you prefer the caller clean the name first, I can add a utility function to the metrics framework and call it in the generated code prior to registration. 

Your thoughts on pushing the responsibility to the caller?

Reply:
Moved the removal to code in ESPCommon that registers the method profiling metric.